lib/ansible/plugins/filter/core.py

```diff
@@ -311,31 +311,41 @@ def combine(*terms, **kwargs):
     if kwargs:
         raise AnsibleFilterError("'recursive' and 'list_merge' are the only valid keyword arguments")
 
-    # allow the user to do `[dict1, dict2, ...] | combine`
-    dictionaries = flatten(terms, levels=1)
+    # allow the user to do `[dict1, dict2, ...] | combine` or `[list1, list2, ...] | combine`
+    items = flatten(terms, levels=1)
 
     # recursively check that every elements are defined (for jinja2)
-    recursive_check_defined(dictionaries)
+    recursive_check_defined(items)
 
-    if not dictionaries:
+    if not items:
         return {}
 
-    if len(dictionaries) == 1:
-        return dictionaries[0]
-
-    # merge all the dicts so that the dict at the end of the array have precedence
-    # over the dict at the beginning.
-    # we merge the dicts from the highest to the lowest priority because there is
-    # a huge probability that the lowest priority dict will be the biggest in size
-    # (as the low prio dict will hold the "default" values and the others will be "patches")
-    # and merge_hash create a copy of it's first argument.
-    # so high/right -> low/left is more efficient than low/left -> high/right
-    high_to_low_prio_dict_iterator = reversed(dictionaries)
-    result = next(high_to_low_prio_dict_iterator)
-    for dictionary in high_to_low_prio_dict_iterator:
-        result = merge_hash(dictionary, result, recursive, list_merge)
-
-    return result
+    if len(items) == 1:
+        return items[0]
+
+    # if all items to be merged are of the type 'list' then merge them together Python style otherwise assume a dict
+    if all(isinstance(item, list) for item in items):
```

Comment:
The `combine` filter doesn't seem like the right place for this to me since the list merging logic is entirely separate. Next, people will want dictionaries in lists to be combined. IMO it would be better off as its own simple filter.

Implementation-wise, it was unintuitive to me that
* combine kwargs are ignored silently
* if the lists being combined are empty, an empty dict is returned
* if there's only one item, the item is returned instead of the list
* combine is not recursive as it is for dictionaries
* lists are flattened, so you need to double-nest, meaning this is an error: `"{{ ['item1'] | combine(['item2']) }}"` because str and str can't be combined (so the example above is wrong, should be `{{ [['item1'], ['item2']] | combine([['item3']]) }}`)
* if the user doesn't realize the point above and this if statement is false because something wasn't double-nested, the error they'll get from vars manager may not help them figure out the format needed for lists - e.g. `"failed to combine variables, expected dicts but got a 'str' and a 'str': \n\"item1\"\n\"item2\""`

Reply:
@s-hertel Thanks for the extensive reply and I agree with all of your points. I also had the struggles of 'should this be combined or not', only I couldn't think of an appropriate name for the potential new filter.

I am willing to refactor this into its own filter. Do you maybe have a suggestion for a possible name? Or should we just go for something simple like `merge_lists` ?

Reply:
That seems okay, though I don't really have a preference. Whatever you pick will probably go through some bikeshedding - naming is difficult.

New content should generally be opened against a collection [(how to figure out where)](https://docs.ansible.com/ansible/latest/community/contributing_maintained_collections.html#deciding-where-your-contribution-belongs). To include the new filter in the Ansible package, the PR should be opened against the community.general collection: https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_contributing.html#contributing-to-a-collection-community-general